Framework/Crystal/src/PredictPeaks.cpp

```diff
@@ -413,13 +413,14 @@ void PredictPeaks::setStructureFactorCalculatorFromSample(
  * @param orientedUB
  * @param goniometerMatrix
  */
-void PredictPeaks::calculateQAndAddToOutput(const V3D &hkl,
+void PredictPeaks::calculateQAndAddToOutput(V3D &hkl,
                                             const DblMatrix &orientedUB,
                                             const DblMatrix &goniometerMatrix) {
   // The q-vector direction of the peak is = goniometer * ub * hkl_vector
   // This is in inelastic convention: momentum transfer of the LATTICE!
   // Also, q does have a 2pi factor = it is equal to 2pi/wavelength.
-  V3D q = orientedUB * hkl * (2.0 * M_PI * m_qConventionFactor);
+  hkl *= m_qConventionFactor;
```

Comment:
Could this also be done in line 435 (`p.setHKL(hkl * m_qConventionFactor);`)? Then `hkl` could still be `const`. But the factor would have to be left in the calculation of `q`.